instrumentation/armeria-1.0/library/src/main/java/io/grpc/override/ContextStorageOverride.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.override;
+
+import com.linecorp.armeria.common.RequestContext;
+import io.grpc.Context;
+import io.grpc.Context.Storage;
+import io.netty.util.AttributeKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ContextStorageOverride extends Storage {
```

Comment:
First, javadoc please :)

Second, having this class will override context storage for the whole JVM with our agent. Including all our instrumentations. And may change the before of the application, if it already has its own override.

Reply:
Yeah - I think this is better behavior for Armeria users than the default. When Armeria context isn't available, the behavior is the same as normal so I'm not terribly worried about affecting the JVM.

But this gRPC API is silly so is problematic... indeed most problematic is if the user wants to override this themselves. Best I can think of is to provide our implementation (I extracted into a separate class) so they can delegate to it.

Reply:
Have you tested this with the full javaagent? I think (shaded) `io.grpc.override.ContextStorageOverride` will be in the AgentClassLoader, while (shaded) `io.grpc.Context` will be in the bootstrap class loader, so may not find the (shaded) `io.grpc.override.ContextStorageOverride`.

Reply:
Yeah I think this is why the `auto` unit tests don't work. I'll continue to follow up on it, but any quick ideas on why the java_concurrent instrumentation wouldn't be enough anyways? https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/979#discussion_r472839958